crates/red_knot_python_semantic/src/types/class.rs

```diff
@@ -1619,4 +1619,52 @@ mod tests {
             );
         }
     }
+
+    #[test]
+    fn known_class_doesnt_fallback_to_unknown_unexpectedly_on_latest_version() {
+        let mut db = setup_db();
+
+        Program::get(&db)
+            .set_python_version(&mut db)
+            .to(PythonVersion::latest());
+
+        for class in KnownClass::iter() {
+            assert_ne!(
+                class.to_instance(&db),
+                Type::unknown(),
+                "Unexpectedly fell back to `Unknown` for `{class:?}`"
+            );
+        }
+    }
+
+    #[test]
+    fn known_class_doesnt_fallback_to_unknown_unexpectedly_on_low_python_version() {
+        let mut db = setup_db();
+
+        Program::get(&db)
+            .set_python_version(&mut db)
+            .to(PythonVersion::PY37);
+
+        for class in KnownClass::iter() {
+            let current_version = Program::get(&db).python_version(&db);
+
+            let version_added = match class {
+                KnownClass::BaseExceptionGroup => PythonVersion::PY311,
+                KnownClass::GenericAlias => PythonVersion::PY39,
+                _ => PythonVersion::PY37,
+            };
+
+            if current_version != version_added {
+                Program::get(&db)
+                    .set_python_version(&mut db)
+                    .to(version_added);
+            }
```

Comment:
Doesn't really matter, but it seems like it would be simpler and more robust just to unconditionally set the version each time through the loop, to the version-added for that class. Does that add noticeable runtime?

Reply:
> Does that add noticeable runtime?

nah, not at all. Not sure what I was thinking there, that's obviously simpler! Thanks!